### PR TITLE
[In-app Purchases] Send app receipt instead of transaction when creating order

### LINKED ISSUE #7931: [In-app Purchases] Send app receipt instead of transaction to the API
After the first receipt validation tests, I understood that you are not supposed to send the Base64-encoded transaction, but the app receipt. This includes refreshing the app receipt if needed.

### PR DESCRIPTION
**Pull request by koke**
Closes: #7931 

### Description

We're supposed to send the app receipt for validation instead of the encoded transaction. This PR does that, ensuring that we refresh the app receipt if missing, since sometimes it won't be there.

I also noticed that we were not handling updates correctly when a transaction was expired/refunded, so I fixed that here as well.

### Testing instructions

1. Go to Hub > IAP Debug
2. Wait for the list of products to load
3. Tap on "1 Month of Debug Woo" and complete the purchase process

The UI is not very helpful, but you should see no errors in the console, and a message like `Successfully registered purchase with Order ID ###`.

Screenshot 2022-10-26 at 10 49 24

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Yosemite/Yosemite/Stores/InAppPurchaseStore.swift [hunk 1]
```diff
@@ -79,15 +79,31 @@ private extension InAppPurchaseStore {
...
 
     func handleCompletedTransaction(_ result: VerificationResult<StoreKit.Transaction>) async throws {
-        switch result {
-        case .verified(let transaction):
-            logInfo("Verified transaction \(transaction.id) (Original ID: \(transaction.originalID)) for product \(transaction.productID)")
-            try await submitTransaction(transaction)
-            await transaction.finish()
-        case .unverified:
+        guard case .verified(let transaction) = result else {
+            // Ignore unverified transactions.
```
Comment (toupper) [inline Yosemite/Yosemite/Stores/InAppPurchaseStore.swift:83]: Will be handle the unverified case in another PR?

**Comment by toupper** [comment]
The code looks and tests well 🎉 ! One thing, I got this message in the Xcode console when testing it:

`[BackgroundTask] Background Task 22 ("SKReceiptRefreshRequest"), was created over 30 seconds ago. In applications running in the background, this creates a risk of termination. Remember to call UIApplication.endBackgroundTask(_:) for your task in a timely manner to avoid this.`

Perhaps we can finish the task so this is not logged?

**Review by toupper — APPROVED** [review]

File: Yosemite/Yosemite/Stores/InAppPurchaseStore.swift (hunk 1 above)
Comment (koke) [reply Yosemite/Yosemite/Stores/InAppPurchaseStore.swift:83]: Honestly, I'm not sure. Apple's [sample code](https://developer.apple.com/documentation/storekit/transaction/3851206-updates) would just ignore those, but I don't fully understand in which scenario can we get an unverified transaction and if it's safe to ignore them.

My understanding is that an unverified transaction would mean that the transaction has been manipulated and can't be trusted, and we'd ignore it unless we want to log that for fraud detection purposes, but I haven't seen enough documentation to back up that understanding.

**Comment by koke** [comment]
> Perhaps we can finish the task so this is not logged

That seems like an internal error and I suspect it happens because there's a retain cycle keeping the `SKReceiptRefreshRequest`. I'll look into it, I don't remember this error on my first attempts
